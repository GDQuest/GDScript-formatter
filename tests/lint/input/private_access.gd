#region Good
class BaseClass:
	var _private_var = 42


	func _private_method():
		pass


	func public_method():
		self._private_method()
		_private_method()


class DerivedClass extends BaseClass:
	func another_method():
		# calling super private method is allowed
		super._private_method()

		_private_var = 100 # accessing super private var is allowed


func good():
	var obj = BaseClass.new()
	obj.public_method()

#endregion

#region Bad

func bad():
	var obj = BaseClass.new()
	obj._private_method()

	obj._private_var = 100

#endregion
