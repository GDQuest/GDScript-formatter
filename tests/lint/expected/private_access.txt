tests/lint/input/private_access.gd:33:private-access:error: Private method '_private_method' should not be called from outside its class
tests/lint/input/private_access.gd:35:private-access:error: Private variable '_private_var' should not be accessed from outside its class